### PR TITLE
[dependencies] update prometheus/client_golang v1.16.0 to v1.18.0

### PR DESCRIPTION
**Pull request by taoyounger**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it:**
This PR updates prometheus/client_golang v1.16.0 to v1.18.0

- Changes in v1.17.0
    - [CHANGE] Minimum required go version is now 1.19 (we also test client_golang against new 1.21 version). https://github.com/prometheus/client_golang/pull/1325
    - [FEATURE] Add support for Created Timestamps in Counters, Summaries and Historams. https://github.com/prometheus/client_golang/pull/1313
    - [ENHANCEMENT] Enable detection of a native histogram without observations. https://github.com/prometheus/client_golang/pull/1314

- Changes in v1.18.0
    - [FEATURE] promlint: Allow creation of custom metric validations. https://github.com/prometheus/client_golang/pull/1311
    - [FEATURE] Go programs using client_golang can be built in wasip1 OS. https://github.com/prometheus/client_golang/pull/1350
    - [BUGFIX] histograms: Add timer to reset ASAP after bucket limiting has happened. https://github.com/prometheus/client_golang/pull/1367
    - [BUGFIX] testutil: Fix comparison of metrics with empty Help strings. https://github.com/prometheus/client_golang/pull/1378
    - [ENHANCEMENT] Improved performance of MetricVec.WithLabelValues(...). https://github.com/prometheus/client_golang/pull/1360

**Which issue(s) this PR fixes:**
Fixes # None

**Special notes for your reviewer:**

**Does this PR introduce a user-facing change?**
```release-note
update prometheus/client_golang v1.16.0 to v1.18.0
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:**

```
None
```

### CODE REVIEW

File: staging/src/k8s.io/component-base/metrics/testutil/metrics_test.go [hunk 1]
```diff
@@ -533,6 +533,7 @@ func TestGetHistogramVecFromGatherer(t *testing.T) {
 						{CumulativeCount: uint64Ptr(1), UpperBound: pointer.Float64Ptr(2.0)},
 						{CumulativeCount: uint64Ptr(1), UpperBound: pointer.Float64Ptr(5.0)},
 					},
+					CreatedTimestamp: timestamp.New(time.Unix(1705044926, 206877000)),
```
Comment (bart0sh) [inline staging/src/k8s.io/component-base/metrics/testutil/metrics_test.go:536]: This seems to cause CI failures as `timestamp` and `time` are not imported.

Comment (ty-dc) [reply staging/src/k8s.io/component-base/metrics/testutil/metrics_test.go]: Thank you, it has been changed

**Review by liggitt — COMMENTED** [review]
mechanical changes look good, one question about the new CreatedTimestamp field and whether we need to do anything with it or if just ignoring it is correct

File: staging/src/k8s.io/component-base/metrics/testutil/metrics_test.go [hunk 2]
```diff
@@ -585,7 +585,7 @@ func TestGetHistogramVecFromGatherer(t *testing.T) {
 			vec.WithLabelValues("value1-1", "value2-1").Observe(4.5)
 			metricName := fmt.Sprintf("%s_%s_%s", HistogramOpts.Namespace, HistogramOpts.Subsystem, HistogramOpts.Name)
 			histogramVec, _ := GetHistogramVecFromGatherer(gather, metricName, tt.lvMap)
-			if diff := cmp.Diff(tt.wantVec, histogramVec, cmpopts.IgnoreFields(dto.Histogram{}, "state", "sizeCache", "unknownFields"), cmpopts.IgnoreFields(dto.Bucket{}, "state", "sizeCache", "unknownFields")); diff != "" {
+			if diff := cmp.Diff(tt.wantVec, histogramVec, cmpopts.IgnoreFields(dto.Histogram{}, "state", "sizeCache", "unknownFields", "CreatedTimestamp"), cmpopts.IgnoreFields(dto.Bucket{}, "state", "sizeCache", "unknownFields")); diff != "" {
```
[the remark this answers is not in the document]
Comment (ty-dc) [reply staging/src/k8s.io/component-base/metrics/testutil/metrics_test.go:588]: Thanks for the reminder. this should be a new feature introduced in v1.17.0.
- prometheus/client_golang#1313

I should not ignore the creation time, but when doing the comparison the following error will appear
```
    --- FAIL: TestGetHistogramVecFromGatherer/filter_with_one_label (0.00s)
        metrics_test.go:595: Got unexpected HistogramVec (-want +got):
              testutil.HistogramVec{
                &{
                        Histogram: &io_prometheus_client.Histogram{
                                ... // 3 ignored and 2 identical fields
                                SampleSum: &1.5,
                                Bucket:    {&{CumulativeCount: &0, UpperBound: &0.5}, &{CumulativeCount: &1, UpperBound: &2}, &{CumulativeCount: &1, UpperBound: &5}},
                                CreatedTimestamp: &timestamppb.Timestamp{
                                        ... // 3 ignored fields
                                        Seconds: 1706235962,
            -                           Nanos:   318936000,
            +                           Nanos:   319122000,
                                },
                                Schema:        nil,
                                ZeroThreshold: nil,
                                ... // 8 identical fields
                        },
                },
```
I tried ignoring "CreatedTimestamp.Nanos", Is it okay to do this? Can you give me some advice? thanks.

**Comment by ty-dc** [comment]
Out of date, a new version has been released.

https://github.com/prometheus/client_golang/releases/tag/v1.19.0

**Pull request closed without merging**